src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -164,6 +164,20 @@ public void drive(Translation2d translation, Rotation2d rotation, boolean fieldO
 		}
 	}
 
+	public void driveWithDoubleSupplierTurn(
+			Translation2d translation, Supplier<Double> rotation, boolean fieldOriented) {
+		// if we're requesting the robot to stay still, lock wheels in X formation
+		if (translation.getNorm() == 0 && rotation.get() == 0.0 && xWheelsEnabled) {
+			swerveDrive.lockPose();
+		}
+		if (rotationSetpoint != null) {
```

Comment:
Make this `else if` as the robot shouldn't try to move while the wheel are in the 'lockPose' position.

Reply:
done